geode-core/src/distributedTest/java/org/apache/geode/cache/ConnectionPoolDUnitTest.java

```diff
@@ -1966,230 +1394,179 @@ public void run2() throws CacheException {
+      Region<Object, Object> region = getRootRegion().getSubregion(name);
+      await().untilAsserted(() -> {
         for (int i = 0; i < 10; i++) {
-          region.put(i, "old" + i);
+          assertThat(region.get("new" + i) == ("callbackArg" + i));
         }
-      }
-    });
-
-    vm2.invoke(createNormal);
-    vm1.invoke("Turn on history", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region = getRootRegion().getSubregion(name);
-        CertifiableTestCacheListener ctl =
-            (CertifiableTestCacheListener) region.getAttributes().getCacheListener();
-        ctl.enableEventHistory();
-      }
-    });
-    vm2.invoke("Update region", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region = getRootRegion().getSubregion(name);
-        for (int i = 0; i < 10; i++) {
-          region.put(i, "new" + i, "callbackArg" + i);
-        }
-      }
-    });
-    Wait.pause(5 * 1000);
+      });
 
-    vm1.invoke("Verify invalidates", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region = getRootRegion().getSubregion(name);
-        CertifiableTestCacheListener ctl =
-            (CertifiableTestCacheListener) region.getAttributes().getCacheListener();
+      CertifiableTestCacheListener<Object, Object> ctl =
+          (CertifiableTestCacheListener<Object, Object>) region.getAttributes()
+              .getCacheListeners()[0];
+      for (int i = 0; i < 10; i++) {
+        Object key = i;
+        ctl.waitForInvalidated(key);
+        await().until(() -> region.getEntry(key) == null);
+      }
+      {
+        List<CacheEvent<Object, Object>> list = ctl.getEventHistory();
+        assertThat(10).isEqualTo(list.size());
         for (int i = 0; i < 10; i++) {
           Object key = i;
-          ctl.waitForInvalidated(key);
-          Region.Entry entry = region.getEntry(key);
-          assertNull(entry); // we are empty!
-        }
-        {
-          List l = ctl.getEventHistory();
-          assertEquals(10, l.size());
-          for (int i = 0; i < 10; i++) {
-            Object key = i;
-            EntryEvent ee = (EntryEvent) l.get(i);
-            assertEquals(key, ee.getKey());
-            assertNull(ee.getOldValue());
-            assertFalse(ee.isOldValueAvailable()); // failure
-            assertEquals(Operation.INVALIDATE, ee.getOperation());
-            assertEquals("callbackArg" + i, ee.getCallbackArgument());
-            assertTrue(ee.isOriginRemote());
-          }
+          EntryEvent ee = (EntryEvent) list.get(i);
+          assertThat(key).isEqualTo(ee.getKey());
+          assertThat(ee.getOldValue()).isNull();
+          assertThat(ee.isOldValueAvailable()).isFalse(); // failure
+          assertThat(Operation.INVALIDATE).isEqualTo(ee.getOperation());
```

Comment:
Couple more assertions to flip around. There are a bunch more in the file(s) but I'm not going to mark them.